src/sinks/util/service/net/tcp.rs

```diff
@@ -0,0 +1,105 @@
+use std::net::SocketAddr;
+
+use snafu::ResultExt;
+use tokio::net::TcpStream;
+
+use vector_config::configurable_component;
+use vector_core::{
+    tcp::TcpKeepaliveConfig,
+    tls::{MaybeTlsSettings, MaybeTlsStream, TlsEnableableConfig},
+};
+
+use crate::dns;
+
+use super::{net_error::*, ConnectorType, HostAndPort, NetError, NetworkConnector};
+
+/// TCP configuration.
+#[configurable_component]
+#[derive(Clone, Debug)]
+pub struct TcpConnectorConfig {
+    /// The address to connect to.
+    ///
+    /// Both IP addresses and hostnames/fully-qualified domain names are accepted formats.
+    ///
+    /// The address _must_ include a port.
+    address: HostAndPort,
```

Comment:
Yeah, fair point.

I'll clean this up.

Reply:
Well, I should say... we'll still probably override the title at the point of usage. For example, if we used this for sinks that _listened_ on an address, then the "connect to" bit would make no sense, and so on.

I can move the description bit to `HostAndPort`, though, since that applies uniformly no matter how it's used, as it relates to the deserialization behavior.

Reply:
Addressed in 74d9bff21.